Auth0/ASProvider.swift

```diff
@@ -1,38 +1,68 @@
         return { url, callback in
-            let session = ASWebAuthenticationSession(url: url, callbackURLScheme: urlScheme) {
-                guard let callbackURL = $0, $1 == nil else {
-                    if let error = $1, case ASWebAuthenticationSessionError.canceledLogin = error {
-                        return callback(.failure(WebAuthError(code: .userCancelled)))
-                    } else if let error = $1 {
-                        return callback(.failure(WebAuthError(code: .other, cause: error)))
-                    }
-
-                    return callback(.failure(WebAuthError(code: .unknown("ASWebAuthenticationSession failed"))))
-                }
+            let session: ASWebAuthenticationSession
 
-                _ = TransactionStore.shared.resume(callbackURL)
+            #if compiler(>=5.10)
+            if #available(iOS 17.4, macOS 14.4, *) {
+                if redirectURL.scheme == "https" {
+                    session = ASWebAuthenticationSession(url: url,
+                                                         callback: .https(host: redirectURL.host!,
+                                                                          path: redirectURL.path),
+                                                         completionHandler: completionHandler(callback))
+                } else {
+                    session = ASWebAuthenticationSession(url: url,
+                                                         callback: .customScheme(redirectURL.scheme!),
+                                                         completionHandler: completionHandler(callback))
+                }
+            } else {
+                session = ASWebAuthenticationSession(url: url,
+                                                     callbackURLScheme: redirectURL.scheme,
+                                                     completionHandler: completionHandler(callback))
             }
+            #else
+            session = ASWebAuthenticationSession(url: url,
+                                                 callbackURLScheme: redirectURL.scheme,
+                                                 completionHandler: completionHandler(callback))
+            #endif
 
             session.prefersEphemeralWebBrowserSession = ephemeralSession
 
             return ASUserAgent(session: session, callback: callback)
         }
     }
 
+    static let completionHandler: (_ callback: @escaping WebAuthProviderCallback) -> ASHandler = { callback in
+        return {
+            guard let callbackURL = $0, $1 == nil else {
+                if let error = $1 as? NSError,
+                    error.userInfo.isEmpty,
+                    case ASWebAuthenticationSessionError.canceledLogin = error {
+                    return callback(.failure(WebAuthError(code: .userCancelled)))
+                } else if let error = $1 {
+                    return callback(.failure(WebAuthError(code: .other, cause: error)))
+                }
+
```

Comment:
Are these bogus code coverage problems? It seems to be complaining about a blank line not being covered.

Reply:
This is due to partial coverage in [this line](https://github.com/auth0/Auth0.swift/blob/2.5.0/Auth0/Request.swift#L57):

<img width="1449" alt="Screenshot 2024-02-29 at 11 06 46" src="https://github.com/auth0/Auth0.swift/assets/5055789/7b6089cc-e117-4ea4-b9e2-fd82491186f0">

Which isn't even relevant anymore, as with Xcode 15 `URLComponents` and `URL` now have the same parsing behavior: https://forums.swift.org/t/url-string-behavior-changed-with-xcode-15-0-beta-5/66570